mocked-api/handlers.ts

```diff
@@ -626,7 +741,7 @@ const cleanupProcessor = (
   // removing properties not needed for the response
   const omitProperties = ["bridge"];
 
-  if (!(processor.filters as Array<Record<string, unknown>>).length) {
+  if (!(processor.filters as Array<Record<string, unknown>>)?.length) {
```

Comment:
@vpellegrino It is right that I can not modify Filteres and Transformation

Reply:
@vpellegrino Finally I was able to change Filteres but only if I changed the name of the procesor. Otherwise I get error `Processor with name 'Processor two' already exists` (in REST response ... not in UI). Is it correct?

Reply:
> @vpellegrino Finally I was able to change Filteres but only if I changed the name of the procesor. Otherwise I get error `Processor with name 'Processor two' already exists` (in REST response ... not in UI). Is it correct?

@manstis could it be connected with your suggestion: `existingProcessors.length === 1 or existingProcessors.length > 0`?

Reply:
ok, looking into this handler, thank you @sjamboro

Reply:
@vpellegrino It most certainly relates to [this](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/54/files/6cae153cea074c1510dbf58e73cf7899f86f5f7e#diff-3904e841d4e956c7e5bec8a10905e9049776263d0e5f502a3062121c42aa0d05R501). If you try to update a processor that already exists the count will be `> 0` (as there is `1` i.e. the one you're editing)... and yes it is caused by my proposal (that fixed the scenario I found, but broke it for @sjamboro's scenario). The problem is that it's OK for the name of a `Processor` "update" to be the same as an existing `Processor` **IF** it is the same `Processor` i.e. `updated.id = existing.id`. I think the check for existing processors needs to be a little smarter and check if the processor being updated is the same as the one that exists.

Reply:
ok @manstis thanks for the suggestion. I'm currently working on it